snapcraft/internal/errors.py

```diff
@@ -355,3 +355,13 @@ class InvalidContainerImageInfoError(SnapcraftError):
 
     def __init__(self, image_info):
         super().__init__(image_info=image_info)
+
+
+class PatcherError(SnapcraftError):
+
+    fmt = (
+        '{elf_file!r} cannot be patched to function properly as a classic '
```

Comment:
s/as/in/ (the file is not the snap, it's just part of one). However, this error can actually be multiple errors, and has no "how to fix" side of it. Do you have an idea of how patchelf might fail, and how one might fix the issue? Perhaps we can break this error class up a little.

Reply:
If an error occurs here, I would have little to no idea on how this would be fixed.

Reply:
Also, I am pro for every error message to be reformatted as part of the exception class, but I do not think we need one exception per type of error, unless it is really useful.

Reply:
It's only useful if the fix for the issue changes depending on the error. However, it sounds like we genuinely don't expect an error here and have no idea how it would be fixed, so this seems fine.

Reply:
(although the typo remains)